### ISSUE TITLE
Bump Gateway API to 0.6.2

### ISSUE THREAD

**Issue by shaneutt** [Kong/kubernetes-ingress-controller#2953]
### Problem Statement

With the `v0.6.2` of the [Gateway API](https://github.com/kubernetes-sigs/gateway-api) we have 2 failing conformance tests: https://github.com/kubernetes-sigs/gateway-api/pull/1814#pullrequestreview-1339787127. In order to upgrade the Gateway API, we need to fix those.

### Related issues:

- https://github.com/Kong/kubernetes-ingress-controller/issues/3726
- https://github.com/Kong/kubernetes-ingress-controller/issues/3120

### Acceptance Criteria

- [ ] the KIC passes all Gateway API `v0.6.2` conformance tests.
- [ ] the CHANGELOG.md has been updated

### Notes

~blocked on the `v0.6.2` release of Gateway API: https://github.com/kubernetes-sigs/gateway-api/pull/1814~
blocked by https://github.com/Kong/kubernetes-ingress-controller/issues/3726

**Comment by mflendrich** [comment]
@mlavacca can you confirm that this is already done and can be closed OR groom this issue into a shape that clearly states what is missing in KIC today?

**Comment by shaneutt** [comment]
There is https://github.com/kubernetes-sigs/gateway-api/pull/1814#pullrequestreview-1339787127 still

**Comment by pmalek** [comment]
@shaneutt but this issue treats about v0.6.0, doesn't it?

**Comment by shaneutt** [comment]
Yes, so if we don't wanna capture the things we need to do for `v0.6.2` here, we just need to make sure we capture them in a follow-up.

**Comment by mlavacca** [comment]
> @mlavacca can you confirm that this is already done and can be closed OR groom this issue into a shape that clearly states what is missing in KIC today?

According to what @shaneutt said, I think we may want to adjust this issue to track what's needed for the upcoming v0.6.2 release. I'll adjust the issue accordingly.

**Comment by rainest** [comment]
Can't really do both and we're definitely going to go to the newer one at some point, so closing in favor of https://github.com/Kong/kubernetes-ingress-controller/issues/3990
